docs/source/cpp/arrays.rst

```diff
@@ -223,3 +227,53 @@ to some logical subsequence of the data.  This is done by calling the
 :func:`arrow::Array::Slice` and :func:`arrow::ChunkedArray::Slice` methods,
 respectively.
 
+.. _fromjson_helpers:
+
+FromJSON Helpers
+================
+
+A set of helper functions is provided for concisely creating Arrays and Scalars
+from JSON_ text. These helpers are intended to be used in examples, tests, or
+for quick prototyping and are not intended to be used where performance matters.
+
+.. _JSON: https://en.wikipedia.org/wiki/JSON
```

Comment:
Should this be a ref to rfc8259 instead of wiki?

Reply:
Yeah, I think that's fine. Changed in f3d37dc363db3d03a605cda37d583eba5dfc54ab.